flake.nix

```
@@ -450,6 +460,26 @@
             else {
               xserver.displayManager.sessionPackages = [cfg.package];
             };
+        })
+        (nixpkgs.lib.mkIf (cfg.enable && cfg.withUWSM) {
+          programs.uwsm = {
+            enable = true;
+            waylandCompositors = {
+              niri = {
+                prettyName = "Niri";
+                comment = "Niri compositor managed by UWSM";
+                # Only takes a path, and `niri-session` uses
+                # the niri default systemd units
+                binPath = nixpkgs.lib.getExe (
+                  pkgs.writeShellScriptBin "niriSession" ''
+                    exec /run/current-system/sw/bin/niri --session
+                  ''
+                );
```

This is what I have in my config, not sure if this is intended behavior or not, but it does work and would simplify this a bit:
```suggestion
                binPath = "/run/current-system/sw/bin/niri -- --session";
```